components/automate-ui/src/app/entities/rules/rule.actions.ts

```diff
@@ -0,0 +1,134 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Action } from '@ngrx/store';
+
+import { Rule } from './rule.model';
+
+export enum RuleActionTypes {
+  GET_ALL         = 'RULES::GET_ALL',
+  GET_ALL_SUCCESS = 'RULES::GET_ALL::SUCCESS',
+  GET_ALL_FAILURE = 'RULES::GET_ALL::FAILURE',
+  GET             = 'RULES::GET',
+  GET_SUCCESS     = 'RULES::GET::SUCCESS',
+  GET_FAILURE     = 'RULES::GET::FAILURE',
+  CREATE          = 'RULES::CREATE',
+  CREATE_SUCCESS  = 'RULES::CREATE::SUCCESS',
+  CREATE_FAILURE  = 'RULES::CREATE::FAILURE',
+  DELETE          = 'RULES::DELETE',
+  DELETE_SUCCESS  = 'RULES::DELETE::SUCCESS',
+  DELETE_FAILURE  = 'RULES::DELETE::FAILURE',
+  UPDATE          = 'RULES::UPDATE',
+  UPDATE_SUCCESS  = 'RULES::UPDATE::SUCCESS',
+  UPDATE_FAILURE  = 'RULES::UPDATE::FAILURE'
+}
```

Comment:
I thought about creating a separate set of Rule actions/reducer/requests/etc for the projects-list rule updates stuff, but ended up just sticking them along with [the existing Project actions](https://github.com/chef/automate/pull/511/files#diff-77f2a4bd0e110812355e14cafa2cae8fR22). Though I think the actions I've added would belong in this set of actions you've added.

Should we tweak the naming to be a little more specific? Like `IngestRule` or `ProjectRule` or something? I only ask there's already an existing concept of `Rule` in the UI for "notification rules": https://github.com/chef/automate/blob/2de87fde778344f37a0e6d10a115175657775bcd/components/automate-ui/src/app/services/rules/rules.service.ts.

Reply:
Yeah I'd second adding these to project actions, since rules can't exist without projects. 

On the API we use `ProjectRule`